hw11/src/main.test.js

```diff
@@ -28,33 +30,77 @@ describe('getAdultUsers', () => {
     it('should return users older than 18', () => {
         const testUsers = [{age: 20}, {age: 17}];
 
-        expect(getAdultUsers(testUsers)).to.be.an('array')
+        expect(getAdultUsers(testUsers))
+            .to.be.an('array')
             .that.includes(testUsers[0]).and
             .that.not.includes(testUsers[1]);
 
     });    
 });
 
 describe('getRandomUsers', () => {
+    const testUsers = [{age: 20}, {age: 17}, {age: 21}, {age: 18}];
+    const mid = 2;
+    let fakeRand, fakeRound;
 
+    beforeEach(() => {
+        fakeRand = sinon.stub(Math, 'random');
+        fakeRound = sinon.stub(Math, 'round');
+    })
+    
+    afterEach(() => {
+        fakeRand.restore();
+        fakeRound.restore();
+    });
+    
+    
     it('should return false if no users', () => {
         expect(getRandomUsers()).to.be.false;
-
     });
     
     it('should return from 0 to 1', () => {
-        const testUsers = [{age: 20}, {age: 17}, {age: 21}, {age: 18}];
-        sinon.stub(Math, 'random').returns(0.7);
+        fakeRand.returns(0.8);
+        fakeRound.returns(mid);
+        expect(getRandomUsers(testUsers))
+            .to.be.an('array')
+            .that.have.members(testUsers.slice(0, mid))
+            .that.not.have.members(testUsers.slice(mid, testUsers.length));
+    });
 
-        expect(getRandomUsers(testUsers)).to.be.an('array').that.includes(testUsers[0], testUsers[1]);        
+    it('should return from 2 to 3', () => {
+        fakeRand.returns(0.1);
+        fakeRound.returns(mid);
+        expect(getRandomUsers(testUsers))
+            .to.be.an('array')
+            .that.have.members(testUsers.slice(mid, testUsers.length))
+            .that.not.have.members(testUsers.slice(0, mid));
+    });
+});
+
+describe('Product', () => {
+    let product;
 
+    beforeEach(() => {
+        product = new Product;
     });
 
-    it('should return from 2 to 3', () => {
-        const testUsers = [{age: 20}, {age: 17}, {age: 21}, {age: 18}];
-        sinon.stub(Math, 'random').returns(0.1);
+    it('should throw error if no value', () => {
+        expect(() => { product.setPrice() }).to.throw(Error);
```

Comment:
why need additional function? Just pass `product.setPrice`

Reply:
I've tried in this way, but it doesn't work.